src/import/name-parser.ts

```diff
@@ -154,19 +146,16 @@ export class BibLatexNameParser {
         return result
     }
 
-    processFirstMiddle(parts /*: Array<string> */) {
+    processFirstMiddle(parts: string[]): void {
         this.nameDict["given"] = this._reformLiteral(parts.join(" ").trim())
     }
 
-    processVonLast(
-        parts /*: Array<string> */,
-        lineage /*: Array<string> */ = []
-    ) {
+    processVonLast(parts: string[], lineage: string[] = []): void {
         let rSplit = this.rsplitAt(parts)
         let von = rSplit[0]
         let last = rSplit[1]
         if (von && !last) {
-            last.push(von.pop())
+            last.push(von.pop()!)
```

Comment:
The idea is if there is a `von` but `last === []` , then it will take the last component of the von and interpret it as the last name.

Reply:
At least I think that is what it is supposed to do. I agree it looks like there is a bug there. as `[]` will not be interpreted as a `falsy` value.

Reply:
I suspect it should be:

```if (von && !last.length) {```

Reply:
That makes sense. I changed the condition to `von.length > 0 && last.length === 0`. That should work

Reply:
Oh, sorry. Didn't see you response

Reply:
That fix results in failed tests. Let me know, how you want to handle/fix this

Reply:
That is because the test output has changed. Do this:

1. In `test/fixtures-import.ts` activate line 42 that goes 'fs.writeFileSync(expected, JSON.stringify(found, null, 4) + "\n")'
2. Run the tests.
3. Comment out the line you activated under 1.
4. Run the file formatting command `npm run format`.
5. Commit.

Reply:
I did that, but I am not sure that the new or old behavior was correct in some cases:

- Cut off citation should not parse: Neither the old behavior of prefix "Smol ci'c" nor the new prefix "Smol", family "ci'c" seems right to me
- Paglione BibTeX should parse: Same as above
- Persian should work: I don't know Persian; this may now be correct

Those were the tests that changed.

Reply:
I know, but the tests files we are using a real-world files that @retorque collected over the years. Some of them have common mistakes in them. So we don't aim at parsing all of them if the syntax is incorrect, but we should not crash and overall have a reasonable success rate.

In this case, I am not sure, so I created this ticket last night: https://github.com/fiduswriter/biblatex-csl-converter/issues/124

Reply:
@DerDrodt Given the answer in https://github.com/fiduswriter/biblatex-csl-converter/issues/124, I think the new test result is perfectly fine.